src/main/java/gift/service/ProductService.java

```diff
@@ -34,39 +40,39 @@ public ProductResponseDto saveProduct(ProductRequestDto dto) {
     }
 
     public ProductResponseDto updateProduct(Long productId, ProductRequestDto dto) {
-        Product product = productRepository.findById(productId);
-        if(product == null) {
-            throw new ProductNotFoundException();
+        if(dto.name() == null) {
+            throw new InvalidNameException();
         }
-        product.updateProductInfo(dto.name(), dto.price(), dto.imageUrl());
-        return product.toDto();
+        if(dto.price() < 0) {
+            throw new InvalidPriceException();
+        }
+        if(dto.imageUrl() == null) {
+            throw new InvalidImageUrlException();
+        }
```

Comment:
ProductRequestDto를 생성할 때 name, price, imageURl을 검증 할 수 있을 것 같아요.
"객체"라는 것을 사용하면 좋아요.
예를 들어서, 다음과 비슷하게 만들 수 있지 않을까요?

```suggestion
public record ProductCreateRequest(String name, Long price, String imageUrl) {
    public ProductCreateRequest {
        if (name == null || name.isBlank()) {
            throw new IllegalArgumentException("상품 이름은 필수입니다.");
        }
        if (price == null || price <= 0) {
            throw new IllegalArgumentException("상품 가격은 0보다 커야 합니다.");
        }
        if (imageUrl == null || imageUrl.isBlank()) {
            throw new IllegalArgumentException("상품 이미지 URL은 필수입니다.");
        }
    }
```